index.ts

```diff
@@ -0,0 +1,45 @@
+interface IUnderlyingMap<K, V> {
+	has(key: K): boolean
+	get(key: K): V | undefined
+	delete(key: K): boolean
+	set(key: K, value: V): void
+}
+
+interface IManagedMap<K, V, GR, U extends IUnderlyingMap<K, V>> {
+	has(key: K): boolean
+	get(key: K): GR
+	delete(key: K): boolean
+	set(key: K, value: V): void
+	getUnderlyingDataStructure(): U
+}
+
+type ValueFactory<K, V> = (key: K) => V
+
+function mapFactory<K, V, U extends IUnderlyingMap<K, V> = Map<K, V>>(
+	factory?: undefined,
```

Comment:
I know I said I didn't want any code changes, but looking at this, I don't know why I ever added support for the factory function being undefined.  I'm cool with dropping that and requiring a factory function.